packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -88,7 +139,142 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
         });
     }
 
-    private _getParamsScript(initializationData: PaypalCommerceInitializationData, cart: Cart): PaypalCommerceScriptParams {
+    private async _onHostedMethodApprove(data: ApproveDataOptions, actions: ApproveActions) {
+        try {
+            const orderDetails = await actions.order.get();
+            const consignments = this._store.getState().consignments.getConsignmentsOrThrow();
+            const lineItems = this._getLineItems();
+            if (!this._paymentMethod?.id) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+            const methodId = this._paymentMethod?.id;
+            const shippingAddress = {
+                ...this._currentShippingAddress,
+                firstName: orderDetails.payer.name.given_name,
+                lastName: orderDetails.payer.name.surname,
+                email: orderDetails.payer.email_address,
+                address1: orderDetails.purchase_units[0].shipping.address.address_line_1,
+            };
+            const consignment = {
+                id: consignments[0].id,
+                shippingAddress,
+                lineItems,
+            };
+            await this._store.dispatch(this._billingAddressActionCreator.updateAddress(shippingAddress));
+
+            await this._store.dispatch(this._consignmentActionCreator.updateConsignment(consignment));
+
+            const submitOrderPayload = {};
+            const submitOrderOptions = {
+                params: {
+                    methodId,
+                },
+            };
+
+            await this._store.dispatch(this._orderActionCreator.submitOrder(submitOrderPayload, submitOrderOptions));
+
+            const paymentData =  {
+                formattedPayload: {
+                    vault_payment_instrument: null,
+                    set_as_default_stored_instrument: null,
+                    device_info: null,
+                    method_id: methodId,
+                    paypal_account: {
+                        order_id: data.orderID,
+                    },
+                },
+                methodId
+            };
+            await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentData));
+            window.location.assign('/checkout/order-confirmation');
+        } catch (e) {
+            throw new RequestError(e);
+        }
+    }
+
+    private async _onShippingChangeHandler(data: ShippingChangeData, _actions: ApproveActions) {
```

Comment:
As far as I see, we can remove actions property because we don't use it in this method

```suggestion
    private async _onShippingChangeHandler(data: ShippingChangeData) {
```

Reply:
fixed